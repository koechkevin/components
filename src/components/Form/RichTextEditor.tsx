import { faPaperclip } from '@fortawesome/pro-regular-svg-icons';
import { Upload } from 'antd';
import { UploadProps } from 'antd/lib/upload/Upload';
import React from 'react';
import { Editor, EditorProps } from 'react-draft-wysiwyg';
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';

import Icon from '../Icon/Icon';

import styles from './RichTextEditor.module.scss';

interface Props extends EditorProps {
  linkCallback?: any;
  fileUploadProps?: UploadProps;
  hasAttachment?: boolean;
  className?: any;
}

const FileUpload: React.FC<UploadProps> = (props) => {
  const { showUploadList, ...restProps } = props;

  return (
    <Upload showUploadList={false} {...restProps}>
      <Icon icon={faPaperclip} style={{ color: '#979797', cursor: 'pointer' }} />
    </Upload>
  );
};

const RichTextEditor: React.FC<Props> = (props) => {
  const {
    toolbar,
    toolbarStyle,
    editorStyle,
    fileUploadProps,
    toolbarCustomButtons,
    hasAttachment,
    className,
    ...restProps
  } = props;

  const getToolBarCustomButtons = (): React.ReactElement<HTMLElement>[] => {
    if (toolbarCustomButtons) {
      return hasAttachment
        ? [<FileUpload {...fileUploadProps} key="fileUpload" />, ...toolbarCustomButtons]
        : toolbarCustomButtons;
    }

    return hasAttachment ? [<FileUpload {...fileUploadProps} key="fileUpload" />] : [];
  };

  const classNames: string = [styles.richTextStyles, className ? className : ''].join(' ');

  return (
    <div className={classNames}>
      <Editor
        toolbarStyle={{ background: '#f3f3f3', fontSize: '16px', height: '48px' }}
        editorStyle={{
          fontSize: '16px',
          color: '#1d1d1d',
          fontWeight: 'normal',
          lineHeight: '24px',
        }}
        toolbar={{
          options: ['inline', 'textAlign', 'list'],
          inline: { options: ['bold', 'italic', 'underline'] },
          list: { options: ['unordered'] },
        }}
        toolbarCustomButtons={getToolBarCustomButtons()}
        {...restProps}
      />
    </div>
  );
};

export default RichTextEditor;
