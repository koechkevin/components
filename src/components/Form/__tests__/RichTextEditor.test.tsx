import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { EditorProps } from 'react-draft-wysiwyg';
import RichTextEditor from '../RichTextEditor';

describe('Renders rich text editor', () => {
  const props: EditorProps = {
    onChange: jest.fn(),
  };

  const wrapper: ReactWrapper = mount(<RichTextEditor {...props} />);
  it('Renders rich text editor without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate change event in the input', () => {
    const inputField: ReactWrapper = wrapper.find('div.notranslate.public-DraftEditor-content').first();
    expect(inputField.exists()).toEqual(true);

    inputField.simulate('change', { event: { target: { value: 'example' } } });
  });
});
