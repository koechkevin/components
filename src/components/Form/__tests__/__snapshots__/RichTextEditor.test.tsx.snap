// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders rich text editor Renders rich text editor without crushing 1`] = `ReactWrapper {}`;
